src/DetectChanges/BCBreak/ClassBased/EnumCasesChanged.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BackwardCompatibility\DetectChanges\BCBreak\ClassBased;
+
+use Psl\Regex;
+use Roave\BackwardCompatibility\Change;
+use Roave\BackwardCompatibility\Changes;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+use Roave\BetterReflection\Reflection\ReflectionEnum;
+use Roave\BetterReflection\Reflection\ReflectionEnumCase;
+
+use function array_filter;
+use function array_map;
+
+class EnumCasesChanged implements ClassBased
+{
+    public function __invoke(ReflectionClass $fromClass, ReflectionClass $toClass): Changes
+    {
+        $fromEnumName = $fromClass->getName();
+        $fromKind     = $this->kindOf($fromClass);
+        $toKind       = $this->kindOf($toClass);
+
+        if (! $fromClass instanceof ReflectionEnum && ! $toClass instanceof ReflectionEnum) {
+            return Changes::empty();
+        }
+
+        if (! $fromClass instanceof ReflectionEnum && $toClass instanceof ReflectionEnum) {
+            return Changes::fromList(Change::changed($fromKind . ' ' . $fromEnumName . ' became enum'));
```

Comment:
Yes, I think class and enum are different enough that it's very unlikely for there to be a useful case way to make that change that isn't a bc break:

- New doesn't work as you say. Not a bc break if the constructor was private or internal, but checking for those is more complexity.
- De-serialisation doesn't work. Again might not be a BC break if it didn't before but would be something else to check for.
- I think the original reason I assumed it would be a BC break is because it means all cases must be new (unless it's a enum with no cases but that's hard to think of a purpose for), and we said already a new case is a BC break. Maybe there's an exception could be made when the old version had zero cases, but again probably complexity we don't need.
- An enum can't be extended, so a BC break if the class was non-final.

Maybe others.

Reply:
Excellent, thanks for confirming!

And indeed, didn't think of (de-)serialization! Good catch!